### PR TITLE
Handle parametrized jobs

### PR DESCRIPTION
**Pull request by swistak35**
They do not have neither evaluation nor allocation.

### CODE REVIEW

File: lib/nomade/deployer.rb [hunk 1]
```diff
@@ -112,32 +203,36 @@ def _deploy
...
-      @logger.info "Waiting until allocations are no longer pending"
-      allocations = @http.allocations_from_evaluation_request(@evaluation_id)
-      until allocations.all?{|a| a["ClientStatus"] != "pending"}
-        @logger.info "."
-        sleep(2)
-        allocations = @http.allocations_from_evaluation_request(@evaluation_id)
-      end
+        @logger.info "Waiting until allocations are no longer pending"
+        allocations = []
+        until allocations.any? && allocations.all?{|a| a["ClientStatus"] != "pending"}
```
Comment (swistak35) [inline lib/nomade/deployer.rb]:
```suggestion
        until !allocations.empty? && allocations.all?{|a| a["ClientStatus"] != "pending"}
```

File: lib/nomade/http.rb [hunk 2]
```diff
@@ -154,12 +154,34 @@ def plan_job(nomad_job)
       raise
     end
 
+    def dispatch_job(nomad_job, payload_data: nil, payload_metadata: nil)
+      if payload_metadata.class == Array && payload_metadata.empty?
```
Comment (swistak35) [inline lib/nomade/http.rb]: :thinking:

Comment (kaspergrubbe) [reply lib/nomade/http.rb]: Yes that is odd, nicely spotted @swistak35, we should just check for a Hash.

Comment (kaspergrubbe) [reply lib/nomade/http.rb]: I've changed this so it just checks for the class of payload_metadata.

**Pull request merged**